docs/oci-storage.md

```diff
@@ -1,40 +1,149 @@
+  * This is a user-defined secret can have any name and must be referenced in the `GitRepo` resource. 
+
+### GitRepo Configuration Example
+
+If a secret named `ocistorage` exists in a namespace. Fleet automatically uses it as the OCI registry for all GitRepo resources in that namespace, and you don’t need to add `ociRegistrySecret` in every GitRepo.
+
+You can explicitly specify a different secret in the GitRepo. The GitRepo spec is extended with information about the OCI Registry. You have to update your GitRepo resource to include a reference to the secret:
+
+```yaml
+metadata:
+  name: coffee-shop-deployments
+  namespace: fleet-local
+spec:
+  repo: https://github.com/example/coffee-shop-fleet
+  branch: main
+  paths:
+    - ./manifests
+  ociRegistrySecret: ocistorage
+```
+
+The `ociRegistrySecret` references a secret in the same namespace, which contains the address and credentials needed to read/write to the OCI registry. 
 
 ```yaml
-// when ociRegistry is defined Fleet will use oci registry as storage
-ociRegistry:
-    // reference is the OCI registry url.
-    reference: "docker.io/your-user-here"
-    // secret name where the credentials for the OCI registry are.
-    // expects a generic secret with username and password keys set.
-    authSecretName: oci-secret
-    // basicHTTP allows Fleet to use basic http connections to communicate
-    // with the registry (defaults to false)
-    basicHTTP: false
-    // insecureSkipTLS allows connections to the OCI registry
-    // without certs (defaults to false)
-    insecureSkipTLS: false
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ocistorage
+  namespace: fleet-local
+type: fleet.cattle.io/bundle-oci-storage/v1alpha1
+data:
+  reference: <base64-encoded-registry-url>
+  username: <base64-encoded-user>
+  password: <base64-encoded-password>
+  insecureSkipTLS: <base64-encoded-true/false>
+  basicHTTP: <base64-encoded-true/false>
+# Optional
+  agentUsername: <base64-encoded-readonly-user>
+  agentPassword: <base64-encoded-password>
 ```
+:::note
+Only the reference field is required. All other fields are optional. 
+:::
+
+If no username or password is specified, Fleet accesses the registry without authentication. You should follow these best practices:
+
+* Use read-only `agentUsername` and `agentPassword` credentials for agents to enhance security.  
+  * However, if you don’t set these credentials, the agent uses user credentials with read/write permissions.  
+* Insecure TLS defaults to `false`, and basic http is disabled by default.   
+  * Fleet allows these flags for development and testing purposes, but they should never be used in production.  
+  * If you use these tags, you expose your app to security vulnerabilities, and put cluster workloads, and credentials at a risk of tampering.  
+* Changing the secret does not update the deployment. The new storage registry is used only after the next Git update or when you trigger a force update.
```

Comment:
This is how Fleet works, right?
Not an item of best practices.... I guess it should go in a different place